crates/types/src/blockchain/block.rs

```diff
@@ -218,8 +220,13 @@ impl PartialFuelBlock {
     ///
     /// Message ids are produced by executed the transactions and collecting
     /// the ids from the receipts of messages outputs.
-    pub fn generate(self, message_ids: &[MessageId]) -> Block {
-        Block::new(self.header, self.transactions, message_ids)
+    pub fn generate(self, message_ids: &[MessageId], event_inbox_root: Bytes32) -> Block {
```

Comment:
Maybe we should clarify that these message ids are for the outbox?

```suggestion
    pub fn generate(self, outbox_message_ids: &[MessageId], event_inbox_root: Bytes32) -> Block {
```